network/wsNetwork.go

```diff
@@ -1165,11 +1212,17 @@ func (wn *WebsocketNetwork) ServeHTTP(response http.ResponseWriter, request *htt
 		prioChallenge:     challenge,
 		createTime:        trackedRequest.created,
 		version:           matchingVersion,
+		identity:          peerPublicKey,
+		identityChallenge: peerIDChallenge,
+		identityVerified:  0,
 		features:          decodePeerFeatures(matchingVersion, request.Header.Get(PeerFeaturesHeader)),
 	}
 	peer.TelemetryGUID = trackedRequest.otherTelemetryGUID
 	peer.init(wn.config, wn.outgoingMessagesBufferSize)
 	wn.addPeer(peer)
+	if shouldSupportIdentityChallenge(matchingVersion) {
+		peer.waitForIdentityVerify()
```

Comment:
If the wait fails and the peer is dropped, is `wn.peersByID` entry removed?
Is there a test for this?

Reply:
Yes -- the peer entry is removed from the map when a peer is dropped, it does it at the same time as when it removes the peer from the peers list.

No -- a test confirming this is not implemented, but can and will be! Thank you for pointing it out :)